ouroboros-network/demo/chain-sync.hs

```diff
@@ -302,27 +304,36 @@ clientBlockFetch sockAddrs = withIOManager $ \iocp -> do
                              LocalConnectionId BlockHeader Block IO
         blockFetchPolicy =
             BlockFetchConsensusInterface {
-              readCandidateChains    = readTVar candidateChainsVar
-                                       >>= traverse readTVar,
-              readCurrentChain       = readTVar currentChainVar,
-              readFetchMode          = return FetchModeBulkSync,
-              readFetchedBlocks      = (\h p -> castPoint p `Set.member` h) <$>
-                                         getTestFetchedBlocks blockHeap,
-              readFetchedMaxSlotNo   = foldl' max NoMaxSlotNo .
-                                       map (maxSlotNoFromWithOrigin . pointSlot) .
-                                       Set.elems <$>
-                                       getTestFetchedBlocks blockHeap,
-              addFetchedBlock        = \p b -> addTestFetchedBlock blockHeap
-                                         (castPoint p) (blockHeader b),
+              consensusRefinementPreBlockFetch =
+                  return,
+              readCandidateChains = do
+                  m <- readTVar candidateChainsVar >>= traverse readTVar
+                  -- TODO @js: is this okay?
+                  return $ Map.map (flip AF.AnnotatedAnchoredFragment AF.FragmentComplete) m,
```

Comment:
Hmmm, so does this mean that all candidate chains are by definition complete? Would it be possible to do some sanity check on this?